### PR TITLE
Fix comparisons in apps

### PR DESCRIPTION
**Pull request by nickvergessen**
Found via https://github.com/nextcloud/server/pull/4767

### CODE REVIEW

File: apps/dav/lib/Connector/Sabre/File.php [hunk 1]
```diff
@@ -147,7 +147,7 @@ public function put($data) {
 			// compare expected and actual size
 			if (isset($_SERVER['CONTENT_LENGTH']) && $_SERVER['REQUEST_METHOD'] === 'PUT') {
 				$expected = $_SERVER['CONTENT_LENGTH'];
-				if ($count != $expected) {
+				if ($count !== $expected) {
```
Comment (nickvergessen) [inline apps/dav/lib/Connector/Sabre/File.php]: int + strings are compared, need to fix

File: apps/dav/lib/Connector/Sabre/File.php [hunk 2]
```diff
@@ -410,7 +410,7 @@ private function createFileChunked($data) {
 		if (isset ($_SERVER['REQUEST_METHOD']) && $_SERVER['REQUEST_METHOD'] === 'PUT') {
 			if (isset($_SERVER['CONTENT_LENGTH'])) {
 				$expected = $_SERVER['CONTENT_LENGTH'];
-				if ($bytesWritten != $expected) {
+				if ($bytesWritten !== $expected) {
```
Comment (nickvergessen) [inline apps/dav/lib/Connector/Sabre/File.php]: int + strings are compared, need to fix

**Comment by MorrisJobke** [comment]
I would say this is not the best point in time to do stuff like that. 🙈

**Comment by nickvergessen** [comment]
Yeah I just wanted to have one check passed, so we can run the general tests (info.xml, language and database), but maybe it's easier to add an option to only run those...

**Comment by MorrisJobke** [comment]
Rebased on current master. @nickvergessen Could you please fix the comments you have?

**Review by MorrisJobke — APPROVED** [review]

**Review by MorrisJobke — DISMISSED** [review]
Unit tests fail

File: apps/files_versions/lib/Storage.php [hunk 3]
```diff
@@ -161,7 +161,7 @@ private static function getVersionsSize($user) {
 	 * store a new version of a file.
 	 */
 	public static function store($filename) {
-		if(\OCP\Config::getSystemValue('files_versions', Storage::DEFAULTENABLED)=='true') {
+		if(\OCP\Config::getSystemValue('files_versions', Storage::DEFAULTENABLED)) {
```
Comment (nickvergessen) [inline apps/files_versions/lib/Storage.php]: Comparing bool to string worked before, however this config is not documented anywhere.
@schiessle any idea what other values are and why it exists?

**Comment by nickvergessen** [comment]
Rebased and fixed

**Review by MorrisJobke — APPROVED** [review]
Tested and works :+1:  Code looks good

**Comment by rullzer** [comment]
Needs a rebase...

**Comment by MorrisJobke** [comment]
> Needs a rebase...

Let me try.

**Comment by MorrisJobke** [comment]
Rebased.

**Review by rullzer — APPROVED** [review]

**Comment by nickvergessen** [comment]
Phan fail is unrelated?

**Comment by MorrisJobke** [comment]
> Phan fail is unrelated?

Correct - this is fixed in master.

**Pull request merged**